### PR TITLE
Fix linking to report for staging URLs

### PR DESCRIPTION
**Pull request by jasperhuangg**
### Details
The code [here](https://github.com/Expensify/App/pull/14487/files#diff-810e7a5e5e8692ea5ea69e3b593e7e6abb3992e2682a7aaf89c8615db6840d83R41-R44) handles linking to specific reports and wasn't correctly capturing staging URLs. This updates it to also handle staging URLs.

### Fixed Issues
$ https://github.com/Expensify/App/issues/12594

### Tests
1. Copy the contents of `.env.staging` into `.env`
2. Sign out of all your sessions on OldDot and NewDot in your browser.
3. Run the app (on Desktop, iOS, and Android). Sign with the following credentials:
    - Username: `nonphotoblue@pfortunezk.com`
    - Password: `asdfASDF00`
4. Click the "CHAT WITH YOUR SETUP SPECIALIST" link sent by Concierge.
5. Verify the report opens in the same client, and doesn't open any sign in links.

### Staging QA
1. Sign with the following credentials.
    - Username: `nonphotoblue@pfortunezk.com`
    - Password: `asdfASDF00`
2. Click the "CHAT WITH YOUR SETUP SPECIALIST" link sent by Concierge.
3. Verify the report opens in the same client, and doesn't open any sign in links.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

Only applicable locally on Desktop, iOS, and Android due to CORS errors on Web platforms.

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/31285285/215623664-67064880-0563-4565-b438-bf46636ce8bf.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/31285285/215625654-67c2f504-70b5-4565-916d-8799ce9472f7.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/31285285/215626200-9ced7d3b-a1c3-42e7-a74b-f4ee7ac52779.mov

</details>

### CODE REVIEW

**Review by alex-mechler — APPROVED** [review]

**Comment by alex-mechler** [comment]
> This updates it to also handle NewDot URLs.

Should this say staging?

**Review by alex-mechler — CHANGES_REQUESTED** [review]
Actually, your QA steps should be updated. If the link is clicked in desktop, it should be opened in the desktop app as well. In addition, please provide videos of you testing this

**Comment by Santhosh-Sellavel** [comment]
PR is failing for me, I set up the environment as STAGING in .env.

Still navigates to the old dot 

https://user-images.githubusercontent.com/85645967/214360622-791f163e-54fc-428d-a7f3-fa164148ff4e.mov

@alex-mechler does it passes for you? 

cc: @jasperhuangg

**Comment by jasperhuangg** [comment]
Hey guys sorry, I should have left this on draft, was about to test it but something came up last minute. I'm gonna put this back on draft for the time being.

**Review by Santhosh-Sellavel — COMMENTED** [review]
Adding myself as a reviewer!

**Comment by jasperhuangg** [comment]
I'm testing this locally and I can't seem to get it to navigate me to OldDot (without my changes), it just opens the report in the same client as expected. I'm checking to see if there were any PRs merged that aren't on staging yet (since this behavior still happens on staging).


https://user-images.githubusercontent.com/31285285/215621685-02e33c8c-a80d-4158-9659-449ae10b7c83.mov

**Comment by Santhosh-Sellavel** [comment]
@jasperhuangg try setting environment as staging locally then perform test steps

**Comment by jasperhuangg** [comment]
Figured it out, thanks @Santhosh-Sellavel I thought I had set up my `.env` correctly for staging but turns out it wasn't.

**Comment by jasperhuangg** [comment]
@Santhosh-Sellavel @alex-mechler thanks for your patience, ready for review!

File: src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const AnchorRenderer = (props) => {
     const attrHref = htmlAttribs.href || '';
     const attrPath = lodashGet(Url.getURLObject(attrHref), 'path', '').replace('/', '');
     const internalNewExpensifyPath = (Url.hasSameExpensifyOrigin(attrHref, CONST.NEW_EXPENSIFY_URL) || Url.hasSameExpensifyOrigin(attrHref, CONST.STAGING_NEW_EXPENSIFY_URL)) && attrPath;
-    const internalExpensifyPath = Url.hasSameExpensifyOrigin(attrHref, CONFIG.EXPENSIFY.EXPENSIFY_URL)
+    const internalExpensifyPath = (Url.hasSameExpensifyOrigin(attrHref, CONFIG.EXPENSIFY.EXPENSIFY_URL) || Url.hasSameExpensifyOrigin(attrHref, CONFIG.EXPENSIFY.STAGING_EXPENSIFY_URL))
                                     && !attrPath.startsWith(CONFIG.EXPENSIFY.CONCIERGE_URL_PATHNAME)
                                     && attrPath;
```
Comment (Santhosh-Sellavel) [inline src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js:32]: `(Url.hasSameExpensifyOrigin(attrHref, CONFIG.EXPENSIFY.EXPENSIFY_URL) || Url.hasSameExpensifyOrigin(attrHref, CONFIG.EXPENSIFY.STAGING_EXPENSIFY_URL))`

This check is not DRY, can we compute once and use it everywhere

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM, tests well!

**Comment by jasperhuangg** [comment]
> @jasperhuangg Just asking for confirmation should it open admins chat or 1:1 with a dedicated specialist?

@Santhosh-Sellavel I'm honestly not entirely sure, I just know it's supposed to navigate directly to the report, I'm not changing how the URL is generated (that's handled in the back-end). Perhaps @alex-mechler can answer this?

**Review by alex-mechler — APPROVED** [review]

**Comment by alex-mechler** [comment]
> Based on how there's a message from what seems like a setup specialist I think that behavior is expected though. @alex-mechler can you confirm?

Correct, the communication between the guide (setup specialist)  and the customer will largely be in the #admins room, as then other admins can also see the conversation. It is expected to go to #admins as a result


---

All you @amyevans

**Review by amyevans — APPROVED** [review]
Looks good!

**Pull request merged**